extensions/iceberg/src/main/java/io/deephaven/iceberg/base/IcebergUtils.java

```diff
@@ -27,45 +44,293 @@ public final class IcebergUtils {
      *
      * @param table The {@link Table} to retrieve data files for.
      * @param snapshot The {@link Snapshot} to retrieve data files from.
-     * @param fileIO The {@link FileIO} to use for reading manifest data files.
+     *
      * @return A stream of {@link DataFile} objects.
      */
-    public static Stream<DataFile> getAllDataFiles(
-            @NotNull final Table table,
-            @NotNull final Snapshot snapshot,
-            @NotNull final FileIO fileIO) {
+    public static Stream<DataFile> allDataFiles(@NotNull final Table table, @NotNull final Snapshot snapshot) {
         try {
-            // Retrieve the manifest files from the snapshot
-            final List<ManifestFile> manifestFiles = snapshot.allManifests(fileIO);
-            return manifestFiles.stream()
-                    .peek(manifestFile -> {
-                        if (manifestFile.content() != ManifestContent.DATA) {
-                            throw new TableDataException(
-                                    String.format(
-                                            "%s:%d - only DATA manifest files are currently supported, encountered %s",
-                                            table, snapshot.snapshotId(), manifestFile.content()));
-                        }
-                    })
-                    .flatMap(manifestFile -> {
-                        try {
-                            final ManifestReader<DataFile> reader = ManifestFiles.read(manifestFile, fileIO);
-                            return StreamSupport.stream(reader.spliterator(), false).onClose(() -> {
-                                try {
-                                    reader.close();
-                                } catch (IOException e) {
-                                    throw new UncheckedIOException(e);
-                                }
-                            });
-                        } catch (final RuntimeException e) {
-                            throw new TableDataException(
-                                    String.format("%s:%d:%s - error reading manifest file", table,
-                                            snapshot.snapshotId(), manifestFile),
-                                    e);
-                        }
-                    });
+            return allManifests(table, snapshot).stream()
```

Comment:
I don't think there is value in this outer try / catch either. The only time it will get triggered is when allManifests throws an error; in which case, it seems like you will just be unnecessarily doubling up on `TableDataException`